platform/ios/src/MGLMapView.mm

```diff
@@ -780,6 +790,31 @@ - (UIViewController *)viewControllerForLayoutGuides
 
 - (void)updateConstraints
 {
+    // scale control
+    //
+    [self removeConstraints:self.scaleControlViewConstraints];
+    [self.scaleControlViewConstraints removeAllObjects];
+    
+    [self.scaleControlViewConstraints addObject:
+     [NSLayoutConstraint constraintWithItem:self.scaleBarView
+                                  attribute:NSLayoutAttributeTop
+                                  relatedBy:NSLayoutRelationEqual
+                                     toItem:self
+                                  attribute:NSLayoutAttributeTop
+                                 multiplier:1
+                                   constant:5+self.contentInset.top]];
```

Comment:
I don't see any constraints between the logo and the attribution button.
The maximum width of the scale bar is `screenWidth/2-padding` which might be a bit too much, especially on iPads but I don't think the answer is to constrain it to the compass. It makes positioning the components with `contentInsets` more prone to misplacement.